### PR TITLE
Jaccard result is in a column major format

### PR DESCRIPTION
**Pull request by orazve**
+ example refactoring

### CODE REVIEW

File: cpp/oneapi/dal/algo/jaccard/backend/cpu/vertex_similarity_default_kernel_avx2.hpp [hunk 1]
```diff
@@ -330,9 +330,10 @@ vertex_similarity_result call_jaccard_default_kernel_avx2(
             }
         }
     }
-    vertex_similarity_result res(homogen_table::wrap(first_vertices, 2, number_elements_in_block),
-                                 homogen_table::wrap(jaccard, 1, number_elements_in_block),
-                                 nnz);
+    vertex_similarity_result res(
+        homogen_table::wrap(first_vertices, number_elements_in_block, 2, data_layout::column_major),
+        homogen_table::wrap(jaccard, number_elements_in_block, 1, data_layout::column_major),
+        nnz);
```
Comment (michael-smirnov) [inline cpp/oneapi/dal/algo/jaccard/backend/cpu/vertex_similarity_default_kernel_avx2.hpp:336]: It's confusing for me - why it works. `data_layout::column_major` does not transpose the data or do another complex logic - it just says that the data stored column by column. Why you swapped the number of rows and columns?

File: cpp/oneapi/dal/algo/jaccard/vertex_similarity_types.hpp [hunk 2]
```diff
@@ -67,22 +67,21 @@ class ONEAPI_DAL_EXPORT vertex_similarity_result {
...
+    ///                                   similarity coefficients
+    /// @param [in]   coeffs              The table of size [nonzero_coeff_count x 1] with
+    ///                                   non-zero Jaccard similarity coefficients
+    /// @param [in]   nonzero_coeff_count The number of non-zero Jaccard coefficients
     vertex_similarity_result(const table& vertex_pairs,
                              const table& coeffs,
                              std::int64_t nonzero_coeff_count);
 
-    /// Returns the table of size 1*nonzero_coeff_count with non-zero Jaccard
+    /// Returns the table of size [nonzero_coeff_count x 1] with non-zero Jaccard
```
Comment (michael-smirnov) [inline cpp/oneapi/dal/algo/jaccard/vertex_similarity_types.hpp:80]: The shape of the table does not depend on the layout - it has `n` rows and `m` columns in both row major and column major formats. The only difference is that in rowmajor format adjacent elements correspond to the same row, and in colmajor - to the same column.
That's the right shape of that table?

**Review by bysheva — APPROVED** [review]

File: examples/oneapi/cpp/source/example_util/output_helpers_graph.hpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+void print_vertex_similarity_result(
+    const oneapi::dal::preview::jaccard::vertex_similarity_result &result) {
+    using namespace oneapi::dal;
+    const std::int64_t nonzero_coeff_count = result.get_nonzero_coeff_count();
+    const std::int64_t lines_count = 20;
+    const std::int64_t print_count =
+        (nonzero_coeff_count < lines_count) ? nonzero_coeff_count : lines_count;
+
+    std::vector<std::int64_t> print_range(print_count);
```
Comment (michael-smirnov) [inline examples/oneapi/cpp/source/example_util/output_helpers_graph.hpp:35]: `#include <vector>` at the head of the file? For now, I suppose this dependency goes from the graph component

File: examples/oneapi/cpp/source/example_util/output_helpers_graph.hpp [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+#include <iomanip>
+#include <iostream>
+#include <numeric>
+
+#include "oneapi/dal/table/common.hpp"
+#include "oneapi/dal/algo/jaccard/vertex_similarity_types.hpp"
+#include "oneapi/dal/table/homogen.hpp"
+
+void print_vertex_similarity_result(
```
Comment (michael-smirnov) [inline examples/oneapi/cpp/source/example_util/output_helpers_graph.hpp:27]: We assume that helpers can be reused between several components. 
Can we make it less specific? Now it's targeted only for similarity and only for jaccard algorithm. What if we want to print out the same similarity results for other graph algorithms?

File: cpp/oneapi/dal/algo/jaccard/backend/cpu/vertex_similarity_default_kernel_avx2.hpp (hunk 1 above)
Comment (orazve) [reply cpp/oneapi/dal/algo/jaccard/backend/cpu/vertex_similarity_default_kernel_avx2.hpp:336]: It is the requirement to store the first vertices and the second vertices in a continuous arrays. But they have to be represented as [n x 2] table. Column-major format is a good option for this case.

File: cpp/oneapi/dal/algo/jaccard/vertex_similarity_types.hpp (hunk 2 above)
Comment (orazve) [reply cpp/oneapi/dal/algo/jaccard/vertex_similarity_types.hpp:80]: Yes, it is right

File: examples/oneapi/cpp/source/example_util/output_helpers_graph.hpp (hunk 3 above)
Comment (orazve) [reply examples/oneapi/cpp/source/example_util/output_helpers_graph.hpp:36]: Added

**Pull request merged**